### DISCUSSION TITLE
heatmap labels of the cells clusters are squished

### DISCUSSION THREAD

**Discussion by jjyotikataria** [reactome/ReactomeGSA#33] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/58770228/125207738-8430ae80-e2ab-11eb-882e-5d338064eacf.png)

@nturaga  @hawrobin  @teslajoy  @jweiser  @SolomonShorser-OICR 

Hi everyone, could you please possibly help me out to fix the label spacing issue of the clusters shown by reactomegsa for the single cell data. As shown in the figure, the data label on the x axis is cropped.

**Comment by jgriss** [comment, marked as answer]
Hi @jjyotikataria,

Any additional parameter passed to the `plot_gsva_heatmap` function is directly forwarded to the `heatmap.2` function.

Unfortunately, even though it's quite powerful, it's still not a great plotting function. Here is the example taken from the single cell analysis vignette showing you how to adapt the main parameters: 

```r
plot_gsva_heatmap(gsva_result, 
                  pathway_ids = relevant_pathways, # limit to these pathways
                  margins = c(6,30), # adapt the figure margins in heatmap.2
                  dendrogram = "col", # only plot column dendrogram
                  scale = "row", # scale for each pathway
                  key = FALSE, # don't display the color key
                  lwid=c(0.1,4)) # remove the white space on the left
```

Essentially, you'll have to play with the `margins` and `lwid` parameters to get the settings right for your plot.

More details on the `heatmap.2` parameters can be found [here](https://earlglynn.github.io/RNotes/package/gplots/heatmap2.html).

**Reply by jjyotikataria** [reply]
@jgriss It worked!!! You saved my time. Thanks a lot.!!!
